### ISSUE TITLE
TF2: Memory leak during implicit conversion from EagerTensor to numpy

### ISSUE THREAD

**Issue by ivankreso** [tensorflow/tensorflow#30873]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tensorflow-gpu==2.0.0-beta1
- Python version: 3.7
- CUDA/cuDNN version: 10.0 / 7.5
- GPU model and memory: GTX 1080Ti

Running the following code on GPU causes a memory leak on my machine during implicit conversion to numpy. Leak does not manifest when doing explicit conversion (`r = np.sum(x.numpy())`).

**Code to reproduce the issue**

```
import tensorflow as tf
import numpy as np

N = 100000

x = tf.zeros((10000)).gpu()
for i in range(N):
    r = np.sum(x)
```

**Comment by anush-o** [comment]
Was able to reproduce the issue with TF version gpu==2.0.0-beta1.Thanks!

**Comment by superbobry** [comment]
Thanks for reporting that @ivankreso! I think the fix (d5b287d6c93332ba73b99b375bd21f81266e3112) did not make it into the -beta1 releases. Could you try with the latest nightly and let us know if it works OK?

```
$ pip install tf-nightly-gpu-2.0-preview
```

**Comment by ivankreso** [comment]
It works ok with the latest nightly version. Thanks.
